@codex please fix

```
> dspace@3.0.0 format:check
> prettier --config .prettierrc --check "**/*.{js,ts,svelte,json,md}"

Checking formatting...
 ✓ tests/eslintPluginPresence.test.ts (1 test) 6ms
 ✓ frontend/tests/gameStateFlush.test.ts (1 test) 5ms
 ✓ tests/checkoutFetchDepth.test.ts (1 test) 10ms
 ✓ tests/npmTestRunsAllSuites.test.ts (1 test) 5ms
 ✓ tests/astroConfigLogLevel.test.ts (1 test) 5ms
 ✓ tests/prismSvelte.test.ts (1 test) 19ms
 ✓ scripts/tests/dbBenchmarkScript.test.ts (1 test) 6ms
 ✓ tests/progressionBalance.test.ts (1 test) 42ms
 ✓ tests/domException.test.ts (1 test) 5ms
 ✓ tests/indexeddb.server.test.ts (1 test) 257ms
 ✓ frontend/tests/sanity.test.ts (1 test) 5ms
[warn] src/pages/processes/__tests__/Processes.spec.ts
[warn] Code style issues found in the above file. Run Prettier with --write to fix.
 ❯ scripts/tests/prettierFormatting.test.ts (1 test | 1 failed) 18577ms
   × Prettier formatting > frontend sources are formatted 18575ms
     → Command failed: npm run format:check

⎯⎯⎯⎯⎯⎯⎯ Failed Tests 2 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  tests/run-test-groups.test.ts > run-test-groups > includes every e2e spec in the configured groups
AssertionError: expected [ Array(1) ] to deeply equal []

- Expected
+ Received

- []
+ [
+   "processes-metadata-loading.spec.ts",
+ ]

 ❯ tests/run-test-groups.test.ts:116:26
    114|     );
    115| 
    116|     expect(missingFiles).toEqual([]);
       |                          ^
    117|   });
    118| 

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/2]⎯

 FAIL  scripts/tests/prettierFormatting.test.ts > Prettier formatting > frontend sources are formatted
Error: Command failed: npm run format:check
 ❯ scripts/tests/prettierFormatting.test.ts:8:5
      6|   test('frontend sources are formatted', () => {
      7|     const frontendDir = path.join(__dirname, '../../frontend');

      8|     execSync('npm run format:check', {
       |     ^
      9|       cwd: frontendDir,
     10|       stdio: 'inherit',

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/2]⎯

 Test Files  2 failed | 289 passed (291)
      Tests  2 failed | 1016 passed (1018)
   Start at  17:51:44
   Duration  164.97s (transform 10.15s, setup 11.56s, collect 45.85s, tests 108.96s, environment 188.15s, prepare 35.22s)


Error: AssertionError: expected [ Array(1) ] to deeply equal []

- Expected
+ Received

- []
+ [
+   "processes-metadata-loading.spec.ts",
+ ]

 ❯ tests/run-test-groups.test.ts:116:26



Error: Error: Command failed: npm run format:check
 ❯ scripts/tests/prettierFormatting.test.ts:8:5

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Serialized Error: { status: 1, signal: null, output: [ null, null, null ], pid: 9353, stdout: null, stderr: null }

 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

new codex task, not a r/e/v/i/e/w task

--- a/frontend/e2e/processes-metadata-loading.spec.ts
+++ b/frontend/e2e/processes-metadata-loading.spec.ts
@@ -47,6 +47,8 @@ test.describe('/processes metadata loading behavior', () => {
     test('does not show preview item ids while metadata is still loading', async ({ page }) => {
         await page.addInitScript(() => {
             // @ts-expect-error test hook
+            window.__DSPACE_ENABLE_TEST_HOOKS__ = true;
+            // @ts-expect-error test hook
             window.__DSPACE_PROCESS_METADATA_DELAY_MS__ = 1200;
         });
 

--- a/frontend/e2e/processes-metadata-loading.spec.ts
+++ b/frontend/e2e/processes-metadata-loading.spec.ts
@@ -45,14 +45,9 @@ test.describe('/processes metadata loading behavior', () => {
     });
 
     test('does not show preview item ids while metadata is still loading', async ({ page }) => {
-        await page.addInitScript(() => {
-            // @ts-expect-error test hook
-            window.__DSPACE_ENABLE_TEST_HOOKS__ = true;
-            // @ts-expect-error test hook
-            window.__DSPACE_PROCESS_METADATA_DELAY_MS__ = 1200;
-        });
-
-        await page.goto('/processes');
+        await page.goto(
+            '/processes?__dspace_enable_test_hooks=1&__dspace_process_metadata_delay_ms=6000'
+        );
         await waitForHydration(page, '.processes-page');
 
         const processRow = page.locator(`[data-process-id="${PROCESS_ID}"]`);
@@ -68,7 +63,7 @@ test.describe('/processes metadata loading behavior', () => {
         await expect(pendingPreviewLine).toContainText(/^(\d+(?:\.\d+)?)x\s*$/);
 
         await expect(pendingPreviewLine).not.toContainText(/^(\d+(?:\.\d+)?)x\s*$/, {
-            timeout: 2500,
+            timeout: 8000,
         });
         await expect(processRow).not.toContainText(PREVIEW_ITEM_ID);
     });

--- a/frontend/scripts/run-test-groups.mjs
+++ b/frontend/scripts/run-test-groups.mjs
@@ -175,7 +175,11 @@ const TEST_GROUPS = [
     },
     {
         name: 'Process Preview',
-        files: ['process-preview.spec.ts', 'process-hardening.spec.ts'],
+        files: [
+            'process-preview.spec.ts',
+            'process-hardening.spec.ts',
+            'processes-metadata-loading.spec.ts',
+        ],
         parallel: true,
         workers: 1,
     },

--- a/frontend/src/pages/processes/Processes.svelte
+++ b/frontend/src/pages/processes/Processes.svelte
@@ -102,7 +102,12 @@
     };
 
     const getMetadataDelayMs = () => {
-        if (import.meta.env.PROD || typeof window === 'undefined') {
+        if (typeof window === 'undefined') {
+            return 0;
+        }
+
+        const hasTestHookAccess = window.__DSPACE_ENABLE_TEST_HOOKS__ === true;
+        if (import.meta.env.PROD && !hasTestHookAccess) {
             return 0;
         }
 

--- a/frontend/src/pages/processes/Processes.svelte
+++ b/frontend/src/pages/processes/Processes.svelte
@@ -106,12 +106,17 @@
             return 0;
         }
 
-        const hasTestHookAccess = window.__DSPACE_ENABLE_TEST_HOOKS__ === true;
+        const searchParams = new URLSearchParams(window.location.search);
+        const hasTestHookAccess =
+            window.__DSPACE_ENABLE_TEST_HOOKS__ === true ||
+            searchParams.get('__dspace_enable_test_hooks') === '1';
         if (import.meta.env.PROD && !hasTestHookAccess) {
             return 0;
         }
 
-        const delay = Number(window.__DSPACE_PROCESS_METADATA_DELAY_MS__);
+        const delay =
+            Number(window.__DSPACE_PROCESS_METADATA_DELAY_MS__) ||
+            Number(searchParams.get('__dspace_process_metadata_delay_ms'));
         return Number.isFinite(delay) && delay > 0 ? delay : 0;
     };
 

--- a/frontend/src/pages/processes/__tests__/Processes.spec.ts
+++ b/frontend/src/pages/processes/__tests__/Processes.spec.ts
@@ -198,7 +198,10 @@ describe('Processes list route contract', () => {
         getItemMapMock.mockResolvedValue(
             new Map([
                 ['known-item', { id: 'known-item', name: 'Known Item', image: '/known.png' }],
-                ['missing-item', { id: 'missing-item', name: 'Unknown item', image: '/favicon.ico' }],
+                [
+                    'missing-item',
+                    { id: 'missing-item', name: 'Unknown item', image: '/favicon.ico' },
+                ],
             ])
         );
 
@@ -234,11 +237,11 @@ describe('Processes list route contract', () => {
         let resolveMetadata:
             | ((value: Map<string, { id: string; name: string; image: string }>) => void)
             | undefined;
-        const metadataPromise = new Promise<Map<string, { id: string; name: string; image: string }>>(
-            (resolve) => {
-                resolveMetadata = resolve;
-            }
-        );
+        const metadataPromise = new Promise<
+            Map<string, { id: string; name: string; image: string }>
+        >((resolve) => {
+            resolveMetadata = resolve;
+        });
         getItemMapMock.mockReturnValue(metadataPromise);
 
         render(Processes, {
@@ -276,7 +279,12 @@ describe('Processes list route contract', () => {
         expect(screen.queryByText('1x Pending Item')).toBeNull();
 
         resolveMetadata?.(
-            new Map([['pending-item', { id: 'pending-item', name: 'Pending Item', image: '/pending.png' }]])
+            new Map([
+                [
+                    'pending-item',
+                    { id: 'pending-item', name: 'Pending Item', image: '/pending.png' },
+                ],
+            ])
         );
 
         expect(await screen.findByText('1x Pending Item')).toBeTruthy();

